sql/core/src/test/scala/org/apache/spark/sql/execution/DataSourceScanExecRedactionSuite.scala

```diff
@@ -19,26 +19,35 @@ package org.apache.spark.sql.execution
 import org.apache.hadoop.fs.Path
 
 import org.apache.spark.SparkConf
-import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.{DataFrame, QueryTest}
+import org.apache.spark.sql.execution.datasources.v2.BatchScanExec
+import org.apache.spark.sql.execution.datasources.v2.orc.OrcScan
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.test.SharedSQLContext
 
 /**
  * Suite that tests the redaction of DataSourceScanExec
  */
-class DataSourceScanExecRedactionSuite extends QueryTest with SharedSQLContext {
+abstract class DataSourceScanRedactionTest extends QueryTest with SharedSQLContext {
 
   override protected def sparkConf: SparkConf = super.sparkConf
-    .set("spark.redaction.string.regex", "file:/[\\w_]+")
+    .set("spark.redaction.string.regex", "file:/[\\w_/]+")
```

Comment:
Do we need to change the pattern by adding '/'?

Reply:
Yes I think so. The original test case can pass because of the output is truncated as such form:
```
*********(redacted)/var/folders/ml/1fm8cdqd3v39v56sx8s9v94m0000gp/T/spark-28300...
```
If the temporary path is shorter, the test won't pass.

Reply:
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/105902/testReport/org.apache.spark.sql.execution/DataSourceScanExecRedactionSuite/treeString_is_redacted/
We can that the redacted path is 
```
*********(redacted)@2/target/tmp/spark-4a1fc3c6-9e60-43ab-956b-821ee8a6f803
```
We should include the symbol `-` and `@`

Reply:
Thank you so much for the explanation!